AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Configuration/DataStoreConfiguration+Helper.swift

```diff
@@ -46,10 +81,29 @@ extension DataStoreConfiguration {
                                       syncExpressions: syncExpressions,
                                       authModeStrategy: authModeStrategy)
     }
-
+    #endif
+    
+    #if os(watchOS)
+    /// The default configuration.
+    public static var subscriptionsDisabled: DataStoreConfiguration {
+        .custom(disableSubscriptions: { false })
+    }
```

Comment:
this is a public computed property to reduce the need to type `.custom(disableSubscriptions: { false }`. Naming as proposed in this PR

Reply:
Why don't we make `custom(disableSubscriptions: { false })` to be  watchOS's version of `.default`?

Reply:
I think our main intention is to force watchOS developers to explicitly set whether subsriptions are enabled or disabled, not just the configuration. The first step to this is achieved through removing the default value
```swift
- configuration dataStoreConfiguration: DataStoreConfiguration = .default)
+ configuration dataStoreConfiguration: DataStoreConfiguration) 
```

So the question is then when watchOS developers set the configuration, will it be clear that `.default` will disable subscriptions?
```swift
let datastore = AWSDataStorePlugin(models: AmplifyModels(), configuration: .default)
```

I think having `.subscriptionsDisabled` (or another naming that conveys subscriptions disabled) will achieve the main intention, also giving the developer the chance to click into the doc comments to read more about watchOS development with DataStore
```swift
let datastore = AWSDataStorePlugin(models: AmplifyModels(), configuration: .subscriptionsDisabled)
```